modules/client/src/main/java/org/apache/ignite/internal/client/TcpClientChannel.java

```diff
@@ -511,10 +518,13 @@ private void handlePartitionAssignmentChange(int flags, ClientMessageUnpacker un
     private void handleNotification(long id, ClientMessageUnpacker unpacker, @Nullable Throwable err) {
         // One-shot notification handler - remove immediately.
         CompletableFuture<PayloadInputChannel> handler = notificationHandlers.remove(id);
+
         if (handler == null) {
-            log.error("Unexpected notification ID [remoteAddress=" + cfg.getAddress() + "]: " + id);
+            // Default notification handler. Used to deliver delayed replication acks.
```

Comment:
This looks a bit dirty to me, and might prevent us from extending the protocol in the future. Can we use `notificationHandlers` as usual?

Reply:
No, we can't. A notification might be not delivered depending on server side condition (no-op operation doesn't produce notifications).
Even if we could, I would keep it this way to avoid additional overhead on creating and removing CHM entry without any need.
Server-side delayed acks implemented in the same way, see org.apache.ignite.internal.tx.impl.TxManagerImpl#onReceived